test/jsonapi/plugs/query_parser_test.exs

```diff
@@ -64,12 +64,31 @@ defmodule JSONAPI.QueryParserTest do
     end
   end
 
-  test "parse_filter/2 turns filters key/val pairs" do
+  test "parse_filter/2 returns filters key/val pairs" do
     config = struct(Config, opts: [filter: ~w(name)], view: MyView)
     filter = parse_filter(config, %{"name" => "jason"}).filter
     assert filter[:name] == "jason"
   end
 
+  test "parse_filter/2 handles nested filters" do
+    config = struct(Config, opts: [filter: ~w(author.username)], view: MyView)
+    filter = parse_filter(config, %{"author.username" => "jason"}).filter
+    assert filter[:author][:username] == "jason"
+  end
+
+  test "parse_filter/2 handles nested filters two deep" do
+    config = struct(Config, opts: [filter: ~w(author.top_posts.text)], view: MyView)
+    filter = parse_filter(config, %{"author.top_posts.text" => "some post"}).filter
+    assert filter[:author][:top_posts][:text] == "some post"
+  end
+
+  test "parse_filter/2 handles nested filters with overlap" do
+    config = struct(Config, opts: [filter: ~w(author.username author.id)], view: MyView)
+    filter = parse_filter(config, %{"author.username" => "jason", "author.id" => "123"}).filter
+    assert filter[:author][:username] == "jason"
+    assert filter[:author][:id] == "123"
+  end
```

Comment:
I thought about whether Config could come with a filter other than `[]` but it doesn't look like it